bns-test/test/name_update.test.ts

```diff
@@ -0,0 +1,205 @@
+    "SPMQEKN07D1VHAB8XQV835E3PTY3QWZRZ5H0DM36"
+  ];
+  const alice = addresses[0];
+  const bob = addresses[1];
+  const charlie = addresses[2];
+  const dave = addresses[3];
+
+  const cases = [{
+    namespace: "blockstack",
+    version: 1,
+    salt: "0000",
+    value: 96,
+    namespaceOwner: alice,
+    nameOwner: bob,
+    priceFunction: {
+      buckets: [7, 6, 5, 4, 3, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+      base: 4,
+      coeff: 250,
+      noVowelDiscount: 4,
+      nonAlphaDiscount: 4,
+    },
+    renewalRule: 4294967295,
+    nameImporter: alice,
+    zonefile: "0000",
+  }, {
+    namespace: "id",
+    version: 1,
+    salt: "0000",
+    value: 9600,
+    namespaceOwner: alice,
+    nameOwner: bob,
+    priceFunction: {
+      buckets: [6, 5, 4, 3, 2, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+      base: 4,
+      coeff: 250,
+      noVowelDiscount: 20,
+      nonAlphaDiscount: 20,
+    },
+    renewalRule: 52595,
+    nameImporter: alice,
+    zonefile: "1111",
+  }];
+
+  beforeEach(async () => {
+    const allocations = [
+      { principal: alice, amount: 10_000_000_000 },
+      { principal: bob, amount: 10_000_000 },
+      { principal: charlie, amount: 10_000_000 },
+      { principal: dave, amount: 10_000_000 },
+    ]
+    const binFile = getDefaultBinaryFilePath();
+    const dbFileName = getTempFilePath();
+    provider = await NativeClarityBinProvider.create(allocations, dbFileName, binFile);
+    bns = new BNSClient(provider);
+    await bns.deployContract();
+  });
+
+
+  it("Given an unlaunched namespace 'blockstack', owned by Alice", async () => {
+    let block_height = 2;
+    let namespace_preorder_ttl = 10;
+    let name_preorder_ttl = 10;
+
+
+      var receipt = await bns.namespacePreorder(cases[0].namespace, cases[0].salt, cases[0].value, { sender: cases[0].namespaceOwner });
+      expect(receipt.success).eq(true);
+      expect(receipt.result).include(`${block_height+namespace_preorder_ttl}`);
+      block_height += 1;
+
+      receipt = await bns.namespaceReveal(
+        cases[0].namespace, 
+        cases[0].salt,
+        cases[0].priceFunction, 
+        cases[0].renewalRule, 
+        cases[0].nameImporter, { sender: cases[0].namespaceOwner });
+      expect(receipt.success).eq(true);
+      expect(receipt.result).include('true');
+    // });
+
+    // it("should be possible for Alice to import a name", async () => {
+      receipt = await bns.nameImport(cases[0].namespace, "alice", "4444", { sender: alice });
+      expect(receipt.success).eq(true);
+      expect(receipt.result).include('true');
+    // })
+
+    // it("should be possible for Alice to update her domain", async () => {
+      receipt = await bns.nameUpdate(
+        cases[0].namespace, 
+        "alice", 
+        "5555", { sender: alice });
+      expect(receipt.error).include('1007');
+      expect(receipt.success).eq(false);
+    // });
+
+    // it("should not be possible for Bob to import a name", async () => {
```

Comment:
Can you also add a test here where Alice imports a name for Bob, but checks that Bob cannot update the name?  Can you also test that Alice cannot import two or more names for Bob (i.e. Bob can own at most one name)?

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/2013/commits/4ffdbe0c4992c7560d1430b232f05cd74d76fbd3